kernel/aster-nix/src/fs/ext2/inode.rs

```diff
@@ -626,6 +632,18 @@ impl Inner {
     }
 
     pub fn resize(&mut self, new_size: usize) -> Result<()> {
+        // Shrink case: First zero the gap between the new size and the
+        // next block boundry (or the old size) if the new size is not block-aligned.
+        let old_size = self.file_size();
+        if old_size > new_size && new_size % BLOCK_SIZE > 0 {
+            let gap_size = old_size.min(new_size.align_up(BLOCK_SIZE)) - new_size;
+            if gap_size > 0 {
+                self.page_cache
+                    .pages()
+                    .write_bytes(new_size, &vec![0; gap_size])?;
+            }
+        }
+
         self.inode_impl.resize(new_size)?;
         self.page_cache.pages().resize(new_size)?;
```

Comment:
I prefer the latter one rather than making modifications in `Vmo`:
```rust
/// Resizes current page cache to a target size.
pub fn resize(&self, new_size: usize) -> Result<()> {
    // If the new size is smaller and not page-aligned,
    // first zero the gap between the new size and the
    // next block boundry (or the old size), if such a gap exists.
    let old_size = self.pages.size();
    if old_size > new_size && new_size % PAGE_SIZE > 0 {
        let gap_size = old_size.min(new_size.align_up(PAGE_SIZE)) - new_size;
        if gap_size > 0 {
            self.pages.write_bytes(new_size, &vec![0; gap_size])?;
        }
    }
    self.pages.resize(new_size)
}
```
And replace all self.page_cache.pages().resize(new_size) to `self.page_cache.resize(new_size)`. What do you think?

Reply:
I agree